### DISCUSSION TITLE
Custom 500 page

### DISCUSSION THREAD

**Discussion by mertafor** [vercel/next.js#55498] [App Router]
I've been able to catch global errors by using global-error.tsx and catch all 404 errors through [...not-found]/page.tsx

However, I cannot see any solution for catching 500 errors and displaying a custom page. It's not being mentioned in documentation and I couldn't find any solutions on the net except pages/500.tsx, which shouldn't be the recommended solution because I don't use pages folder at all. 

Is there a way to catch 500 and display a custom page?

**Comment by kubmir** [comment]
Yes, it seems like it is not possible unfortunately. It would be nice to have some way to catch errors while doing server rendering a return custom 500 screen with status 500 from server. It is not ideal that it return 200 and then can be handled only on client component. What are the plans for this ? I have seen several issues regarding this ...

**Comment by delasign** [comment]
This is still an issue on May 7th 2024

**Comment by gurkanyerlikayaoglu** [comment]
This is still an issue on April 10th 2025

**Comment by tarunkhurana** [comment]
This is still an issue October 22th 2025
